packages/composite-checkout/src/lib/payment-methods/stripe-credit-card-fields.js

```diff
@@ -527,9 +528,16 @@ function isCreditCardFormValid( store, __ ) {
 
 function CreditCardLabel() {
 	const { __ } = useI18n();
+	const allPaymentMethods = useAllPaymentMethods();
+	const hasExistingCard = allPaymentMethods.filter( ( paymentMethod ) =>
+		paymentMethod.id.startsWith( 'existingCard-' )
```

Comment:
This is a good approach, but `existingCard-` is a [calypo-specific implementation detail](https://github.com/Automattic/wp-calypso/blob/64e64e4a5dcfd3aabec94f9405c9728c8c61df63/client/my-sites/checkout/composite-checkout/use-create-payment-methods.js#L368). What if a different host called their existing cards `existing-card-`?

We might want to offload this responsibility to the host. `hasExistingCard` could be a boolean argument passed to `createStripeMethod()` and then passed as a prop to this component. Alternatively, we could just replace the label inside calypso [like we did with credits](https://github.com/Automattic/wp-calypso/blob/64e64e4a5dcfd3aabec94f9405c9728c8c61df63/client/my-sites/checkout/composite-checkout/use-create-payment-methods.js#L297) and [free purchases](https://github.com/Automattic/wp-calypso/blob/64e64e4a5dcfd3aabec94f9405c9728c8c61df63/client/my-sites/checkout/composite-checkout/use-create-payment-methods.js#L314). I'd probably start with that approach since it's an already-established pattern and wouldn't involve changing anything in the `composite-checkout` package.